go/pkg/cas/upload.go

```diff
@@ -108,14 +114,53 @@ type UploadOptions struct {
 // result in a dangling symlink.
 var ErrSkip = errors.New("skip file")
 
+// UploadResult is the result of a Client.Upload call.
+// It provides file/dir digests and statistics.
+type UploadResult struct {
+	Stats TransferStats
+	u     *uploader
+}
+
+// Digest returns the digest computed for a file/dir at ps.Path.
+//
+// To retrieve a digest of a regular file, only ps.Path is required.
```

Comment:
I'd go as far as saying "only ps.Path is checked - other fields are ignored" (pending I haven't checked your work on Allowlists yet)

Reply:
done